README.md

```diff
@@ -195,11 +195,114 @@ Subscribe to the topic published by the AWS IoT thing in the client, to see mock
+        "iot:Receive"
+      ],
+      "Resource": [
+        "arn:aws:iot:REGION:ACCOUNT_ID:topic/$aws/things/THING_NAME/shadow/name/SHADOW_NAME/update/documents"
+      ]
+    },
+    {
+      "Effect": "Allow",
+      "Action": [
+        "iot:Publish"
+      ],
+      "Resource": [
+        "arn:aws:iot:REGION:ACCOUNT_ID:topic/$aws/things/THING_NAME/shadow/name/SHADOW_NAME/update"
+      ]
+    }
+  ]
+}
+```
+
+This policy allows the IoT Thing to connect to the named shadow, publish to it, and subscribe to its update. The policy must be created and attached to the certificate used for the Thing. This can be done by copying the template and replacing the `REGION`, `ACCOUNT_ID`, `THING_NAME`, and `SHADOW_NAME` fields, then creating and attaching the policy. The commands are as follows:
+
+```bash
+export SHADOW_POLICY_TEMPLATE=~/aws-iot-robot-connectivity-samples-ros2/templates/shadow_policy_template.json
+export SHADOW_POLICY_FILE=~/aws-iot-robot-connectivity-samples-ros2/iot_certs_and_config/shadow_policy.json
+cat $SHADOW_POLICY_TEMPLATE > $SHADOW_POLICY_FILE
+sed -i -e "s/REGION/$AWS_REGION/g" $SHADOW_POLICY_FILE
+sed -i -e "s/ACCOUNT_ID/$ACCOUNT_ID/g" $SHADOW_POLICY_FILE
+sed -i -e "s/THING_NAME/$THING_NAME/g" $SHADOW_POLICY_FILE
+sed -i -e "s/SHADOW_NAME/$SHADOW_NAME/g" $SHADOW_POLICY_FILE
+export SHADOW_POLICY_NAME=ros2_shadow_policy
+aws iot create-policy --policy-name $SHADOW_POLICY_NAME --policy-document file://$SHADOW_POLICY_FILE
+aws iot attach-policy --policy-name $SHADOW_POLICY_NAME --target $CERT_ARN
+```
+
+This is sufficient permission to interact with the shadow. The sample application can now be run as follows:
+
+```bash
+source ~/aws-iot-robot-connectivity-samples-ros2/workspace/install/setup.bash
+# Replace tilde with home directory
+export FULL_CONFIG_PATH="${IOT_CONFIG_FILE/#\~/$HOME}"
+ros2 launch iot_shadow_service crack_safe.yaml shadow_name:=$SHADOW_NAME path_for_config:=$FULL_CONFIG_PATH
+```
+
+To view the named shadow changing, navigate to the IoT Core page of the AWS console, as shown below:
+![IoT search bar](images/iot_core_console.png)
+
+Select the Things page from All Devices on the navigation bar on the left:
+![IoT Things Menu](images/iot_things_menu.png)
+
+Select the Thing from the list of Things that is presented:
+![IoT Things List](images/iot_thing.png)
+
+From the details page, select the Device Shadow tab:
+![IoT Thing Details](images/iot_thing_details.png)
+
+Select the named shadow:
+![Named Shadow](images/named_shadow.png)
+
+Scroll down a little to the Shadow document. The numbers in this state should be updating with the sample application updating them. The document looks like this:
+![Named Shadow Document](images/named_shadow_state.png)
+
+You can now write updates to your IoT Core Named Shadow from a ROS2 service call, and publish any updates to the Shadow published to a ROS2 topic, allowing connected ROS2 nodes to benefit from a human-readable JSON format document synchronized with AWS.
```

Comment:
For closing the loop with the user, could you add a couple of lines on how customers can see this data flowing in ROS2? Which topics they can subscribe to and a screenshot on what it looks like .

Reply:
Agreed - coming in next commit